### ISSUE TITLE
the model export process seems not  properly  under some  case?

### ISSUE THREAD

**Issue by bolt163** [mozilla/DeepSpeech#1410]
The case with me now  was like this: the training process was  break off  for gpu resources were always  scheduling varing to time(the training enviroment for me is  using online resources training，for time at peak period, the gpu resources will be recycled for important  online  work, while for  idle time the gpu resources will be  release back  again for  training job using)， so the training process may  be interrupt times; 

after 3 times' of interrupt ,   the training loss was already converged  to a  proper value, and the FINISHED Optimization log was already print out which indicate the training  process has already been end! while now ,

I STARTING Optimization
I Training of Epoch 12 - loss: 7.242887
I Validation of Epoch 12 - loss: 9.995953
I Training of Epoch 13 - loss: 7.176886
I Validation of Epoch 13 - loss: 9.645760
I Training of Epoch 14 - loss: 7.089039
I Validation of Epoch 14 - loss: 9.557784
I Training of Epoch 15 - loss: 6.914542
I Validation of Epoch 15 - loss: 8.357721
I FINISHED Optimization - training time: 15:58:24

————while at the time of  the  training  process ending, the  deepspeech running job was interrupt  again, and i restart the job again running for export  process (1829 line of DeepSpeech.py) (https://github.com/mozilla/DeepSpeech/blob/master/DeepSpeech.py#L1829 ) for  getting  the  output_graph.pb model,  this step cost several time's of day,  i thought this was not that properly?
(my training epoch cost about 27hours, and there were 3000 hours's traing data, and the test data was about 20hours)

Restart the traing job, it still will go to the  train() process(1795 line of DeepSpeech.py)(https://github.com/mozilla/DeepSpeech/blob/master/DeepSpeech.py#L1795),  the training was already end , so restarting the job, this time the train() process's job was restoring from the checkpoint and do the test process  

【My doubt here is】: while  the train  was already end(log FINISHED Optimization already out) and the export porcess was not ending, if the training was interrupt at this time,  while restart the job  directly  go to the output graph pb model export process ， is there any performance loss with the model? thanks

![_152885911122](https://user-images.githubusercontent.com/25170435/41328801-aa189782-6efc-11e8-93cc-f964fa33d144.png)

**Comment by bolt163** [comment]
solved it myself, and i think there should be  a option  solution for directly export the final pb model  from the checkout point   if training ended  while  the total job  was  interrupt   before  the exporting process ending

i will create a new issue  and give my solution about this latter...

**Comment by bolt163** [comment]
I have compaired  the two models:
  (i)directly export the model out not doing the testing porcess   ——namely Model1 

  (ii)first runing the tesing process on the test set(which may will takes times of days if the test set was large), and then export the model out ——namely Model2

  I  have used the above Model1/Model2 to  test the same corpus, the WER were nearly the same

**Comment by lissyx** [comment]
@bolt163 Thanks, I'm not sure i get the exact issue, since you document about potential loss of quality, and then you document no WER change.

BTW, documenting code behavior with screenshots is painful and not helping understanding the problem :/

**Comment by reuben** [comment]
If you pass `--notrain --notest` with the other flags then it should go directly to the exporting process and take seconds. Does that not work for you?

**Comment by lissyx** [comment]
@bolt163 Code moved, could you please confirm if this is still an issue ?

**Comment by lissyx** [comment]
Closing for lack of activity, and being unable to reproduce.
